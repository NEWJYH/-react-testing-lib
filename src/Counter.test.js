import React from 'react';
import {render, fireEvent} from "@testing-library/react";
import Counter from './Counter';

describe("<Counter />", () => {
    it("matches snapshot", () => {
        const utils = render(<Counter />);
        expect(utils.container).toMatchSnapshot();
    });

    it("has a number and two buttons", () => {
        const utils = render(<Counter />);
        utils.getByText('0');
        utils.getByText('+1');
        utils.getByText('-1');
    });
    
    // 이벤트 처리 
    it("increase", () => {
        const utils = render(<Counter />);
        const number = utils.getByText("0");
        const plusButton = utils.getByText("+1");
        fireEvent.click(plusButton);
        fireEvent.click(plusButton);
        expect(number).toHaveTextContent("2");
    });
    it("decrease", () => {
        const utils = render(<Counter />);
        const number = utils.getByText("0");
        const minusButton = utils.getByText("-1");
        fireEvent.click(minusButton);
        fireEvent.click(minusButton);
        expect(number).toHaveTextContent("-2");
    });
});