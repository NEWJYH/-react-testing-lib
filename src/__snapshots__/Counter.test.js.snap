// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> matches snapshot 1`] = `
<div>
  <div>
    <h1>
      0
    </h1>
    <button>
      +1
    </button>
    <button>
      -1
    </button>
  </div>
</div>
`;
